src/test/java/teammates/sqlui/webapi/RegenerateInstructorKeyActionTest.java

```diff
@@ -0,0 +1,166 @@
+package teammates.sqlui.webapi;
+
+import static org.mockito.Mockito.when;
+
+import java.time.Instant;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.InstructorPrivileges;
+import teammates.common.util.Const;
+import teammates.common.util.Const.InstructorPermissions;
+import teammates.common.util.JsonUtils;
+import teammates.storage.sqlentity.Account;
+import teammates.storage.sqlentity.Course;
+import teammates.storage.sqlentity.FeedbackSession;
+import teammates.storage.sqlentity.Instructor;
+import teammates.ui.output.InstructorData;
+import teammates.ui.request.Intent;
+import teammates.ui.webapi.RegenerateInstructorKeyAction;
+import teammates.common.util.EmailType;
+import teammates.common.util.EmailWrapper;
+import teammates.ui.output.RegenerateKeyData;
+import teammates.ui.webapi.JsonResult;
+
+/**
+ * SUT: {@link RegenerateInstructorKeyAction}.
+ */
+public class RegenerateInstructorKeyActionTest extends BaseActionTest<RegenerateInstructorKeyAction> {
+
+    String googleId = "user-googleId";
+
+    Course course;
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.INSTRUCTOR_KEY;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return POST;
+    }
+
+    @BeforeMethod
+    public void baseClassSetup() {
+        course = new Course("course-id", "name", Const.DEFAULT_TIME_ZONE, "institute");
+        loginAsAdmin();
+    }
+
+    @Test
+    protected void testExecute_notEnoughParameters_throwsInvalidHttpParameterException() {
+        Instructor instructor = new Instructor(course, "name", "email@tm.tmt", false, "", null, null);
+        when(mockLogic.getInstructorByRegistrationKey(instructor.getRegKey())).thenReturn(instructor);
```

Comment:
Issue with null values is because return values are [mocked](https://en.wikipedia.org/wiki/Mock_object). By default, the mocking library we use returns null if we do not specify a value to return.

Don't worry too much about this for now, the unit tests already needs a revamp anyways. Let's focus on migrating the existing tests (which are integration tests) to the integration test package (look for it it package).